storage/remote/queue_manager.go

```diff
@@ -1035,24 +1145,34 @@ func (s *shards) enqueue(ref chunks.HeadSeriesRef, data sampleOrExemplar) bool {
 type queue struct {
 	// batchMtx covers operations appending to or publishing the partial batch.
 	batchMtx   sync.Mutex
-	batch      []sampleOrExemplar
-	batchQueue chan []sampleOrExemplar
+	batch      []timeSeries
+	batchQueue chan []timeSeries
 
 	// Since we know there are a limited number of batches out, using a stack
 	// is easy and safe so a sync.Pool is not necessary.
 	// poolMtx covers adding and removing batches from the batchPool.
 	poolMtx   sync.Mutex
-	batchPool [][]sampleOrExemplar
+	batchPool [][]timeSeries
 }
 
-type sampleOrExemplar struct {
+type timeSeries struct {
```

Comment:
I didn’t want to do “sampleOrExemplarOrHistogram,” that sounds like a little bit too much, so I took the name from the “TimeSeries” proto which holds all three data types. Another possible name could be “toBeSent” or “queued” or something like that.

Reply:
`timeSeries` is reasonable to me, I certainly prefer it to `sampleOrExemplarOrHistogram`.